### PR TITLE
Provide a way for customers to access the internal registry mirror

### PR DESCRIPTION
**Pull request by valentyn88**
Fixes #

## Proposed Changes

  - To allow users to download from a known source, the environment variable `SAUCECTL_INSTALL_BINARY` was added.

### CODE REVIEW

**Review by christian-bromann — COMMENTED** [review]
I think it is worth for this change to add a test for it.

**Comment by valentyn88** [comment]
> I think it is worth for this change to add a test for it.

Sure. I still feel not super confident with nodejs and thanks to @dpgraham that helper me to cover code by unit tests.

File: README.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ To install `saucectl` using NPM just run:
 $ npm i -g saucectl
 ```
 
+In case you need to download `saucectl` from a known source, you can use environment variable:
```
Comment (FriggaHel) [inline README.md]: Should we mention that this needs to be done either at install or first run (if install scripts are disabled) ?

File: index.js [hunk 2]
```diff
@@ -5,17 +5,30 @@ const BinWrapper = require('bin-wrapper');
...
-	.src(`${base}/v${version}/saucectl_${version}_win_32-bit.zip`, 'win32', 'x86')
-	.src(`${base}/v${version}/saucectl_${version}_win_32-bit.zip`, 'win32', 'x64')
-	.src(`${base}/v${version}/saucectl_${version}_win_64-bit.zip`, 'win64', 'x64')
-    .dest(path.join(__dirname, 'bin'))
-    .use(process.platform.startsWith('win') ? 'saucectl.exe' : 'saucectl')
-	.version(`v${version}`);
+const binWrapper = (binInstallURL) => {
+    const bw = new BinWrapper();
+
+    if (binInstallURL) {
```
Comment (FriggaHel) [inline index.js:11]: Should we also check that url is valid (at least that this is really a url?)

**Review by FriggaHel — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -11,6 +11,12 @@ To install `saucectl` using NPM just run:
 $ npm i -g saucectl
 ```
 
+In case you need to download `saucectl` from a known source or run `sauceclt` after installation from a known source, you can use environment variable:
```
Comment (vrunoa) [inline README.md]: `sauceclt` typo. Even tho, I'm a bit confused about what does this implies: 
`run `sauceclt` after installation from a known source`. Do we need this second description?

File: index.js (hunk 2 above)
Comment (valentyn88) [reply index.js:11]: `bin-wrapper` will throw an exception if the URL is not reachable. I edited the message related to this.

File: index.js [hunk 4]
```diff
@@ -5,17 +5,30 @@ const BinWrapper = require('bin-wrapper');
...
-	.src(`${base}/v${version}/saucectl_${version}_mac_64-bit.tar.gz`, 'darwin', 'x64')
-	.src(`${base}/v${version}/saucectl_${version}_linux_32-bit.tar.gz`, 'linux', 'x86')
-	.src(`${base}/v${version}/saucectl_${version}_linux_64-bit.tar.gz`, 'linux', 'x64')
-	.src(`${base}/v${version}/saucectl_${version}_win_32-bit.zip`, 'win32', 'x86')
-	.src(`${base}/v${version}/saucectl_${version}_win_32-bit.zip`, 'win32', 'x64')
-	.src(`${base}/v${version}/saucectl_${version}_win_64-bit.zip`, 'win64', 'x64')
-    .dest(path.join(__dirname, 'bin'))
-    .use(process.platform.startsWith('win') ? 'saucectl.exe' : 'saucectl')
-	.version(`v${version}`);
+const binWrapper = (binInstallURL) => {
```
Comment (vrunoa) [inline index.js]: what do you think about adding a default value? `(binInstallURL = null)`

File: index.js (hunk 2 above)
Comment (vrunoa) [reply index.js:11]: could the if statement only wrap `.src` methods as the other are the same?

File: index.js [hunk 5]
```diff
@@ -5,17 +5,30 @@ const BinWrapper = require('bin-wrapper');
...
+    if (binInstallURL) {
+        bw.src(binInstallURL, '', '')
+            .dest(path.join(__dirname, 'bin'))
+            .use(process.platform.startsWith('win') ? 'saucectl.exe' : 'saucectl');
+        return bw;
+    }
+
+    bw.src(`${base}/v${version}/saucectl_${version}_mac_32-bit.tar.gz`, 'darwin', 'x86')
+        .src(`${base}/v${version}/saucectl_${version}_mac_64-bit.tar.gz`, 'darwin', 'x64')
+        .src(`${base}/v${version}/saucectl_${version}_mac_64-bit.tar.gz`, '', '')
```
Comment (vrunoa) [inline index.js]: @valentyn88 why does this one has an empty platform and arch?

File: README.md (hunk 3 above)
Comment (valentyn88) [reply README.md]: The first version with the only mentioning install was correct.

File: README.md (hunk 1 above)
Comment (valentyn88) [reply README.md]: Why do we need to disable the install script?

File: index.js (hunk 2 above)
Comment (valentyn88) [reply index.js:11]: good idea.

File: index.js (hunk 5 above)
Comment (valentyn88) [reply index.js]: I forgot to delete this line. Now everything is ok.

File: index.js (hunk 4 above)
Comment (valentyn88) [reply index.js]: I added the default value.

File: README.md (hunk 1 above)
Comment (FriggaHel) [reply README.md]: This is the use case of some customers, they disable install scripts for security.
In that case, `SAUCECTL_INSTALL_BINARY` should be passed not-only for `npm install saucectl`, but also the first run of `npx saucectl` / `saucectl` depending how you invoke it.

Comment (valentyn88) [reply README.md]: I didn't know about this case. Thanks for raising this question. I've updated my PR.

**Review by vrunoa — APPROVED** [review]

**Review by tianfeng92 — APPROVED** [review]

**Pull request merged**